### PR TITLE
Added vision renderer effects (including colorblind)

### PR DESCRIPTION
**Pull request by Arkendorf**

### CODE REVIEW

File: Assets/Scripts/RendererFeatures/Blit.cs [hunk 1]
```diff
@@ -37,20 +36,17 @@ public override void Create()
...
-                var src = renderer.cameraColorTarget;
-                var dest = (settings.destination == Target.Color) ? RenderTargetHandle.CameraTarget : m_RenderTextureHandle;
-
-                if (settings.blitMaterial == null)
-                {
-                    Debug.LogWarningFormat("Missing Blit Material. {0} blit pass will not execute. Check for missing reference in the assigned renderer.", GetType().Name);
-                    return;
-                }
+            var src = renderer.cameraColorTarget;
+            var dest = (settings.destination == Target.Color) ? RenderTargetHandle.CameraTarget : m_RenderTextureHandle;
```
Comment (j-mai) [inline Assets/Scripts/RendererFeatures/Blit.cs:40]: what does this mean?

File: Assets/Scripts/Vision/RendererEffect.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering.Universal;
+
+namespace Anivision.Vision
+{
+    // Parent class for effects that modify the render pipeline
+    public abstract class RendererEffect : MonoBehaviour
+    {
+        public VisionEffect Effect { get; } // The vision effect that corresponds to this renderer effect
```
Comment (j-mai) [inline Assets/Scripts/Vision/RendererEffect.cs:12]: maybe you want to make this an abstract, so classes inheriting from this are forced to use override and implement a getter for this.

File: Assets/Scripts/RendererFeatures/Blit.cs (hunk 1 above)
Comment (Arkendorf) [reply Assets/Scripts/RendererFeatures/Blit.cs:40]: The only thing I changed in this script was getting rid of modifications I made previously. Now it is purely the script from the demo project. I had made a change to allow you to only target the main camera with a blit, but that was before I realized there was a built-in way to do it. If you look at this file's diff, you'll see that the lines you commented on were already in the script, I just unindented them.

File: Assets/Scripts/Vision/RendererEffect.cs (hunk 2 above)
Comment (Arkendorf) [reply Assets/Scripts/Vision/RendererEffect.cs:12]: Yeah, I tried that initially, but variables set as override / abstract were not visible in the inspector, so I couldn't make it into a dropdown.

Comment (j-mai) [reply Assets/Scripts/Vision/RendererEffect.cs:12]: Why does it need to be exposed in the inspector? Shouldn't all classes that inherit from this have a set VisionEffect that shouldn't be modified through the inspector? For example, the Colorblind script should always return a VisionEffect of Colorblind, no?

File: Assets/Scripts/RendererFeatures/Blit.cs (hunk 1 above)
Comment (j-mai) [reply Assets/Scripts/RendererFeatures/Blit.cs:40]: ah ok got it

File: Assets/Scripts/Vision/RendererEffect.cs (hunk 2 above)
Comment (Arkendorf) [reply Assets/Scripts/Vision/RendererEffect.cs:12]: Yes, that's true for the most part. BlitEffect.cs is a more general purpose effect that can be modified in the inspector though.

Comment (j-mai) [reply Assets/Scripts/Vision/RendererEffect.cs:12]: oh so `BlitEffect.cs` allows easy customization of a blit effect if people don't want to write their own scripts? that's cool. ok, got it.

Comment (Arkendorf) [reply Assets/Scripts/Vision/RendererEffect.cs:12]: Yes, that was the goal at least

**Pull request merged**